File: ratpack-client/src/main/java/com/gradle/export/client/ExportClientSpec.java
package com.gradle.export.client;

import java.net.URI;
import java.nio.file.Path;

class ExportClientSpec {

    final URI serverUri;
    final String username;
    final String password;
    final int days;
    final Path outputFile;
    final boolean obfuscate;
    final int workers;

    private ExportClientSpec(URI serverUri, String username, String password, int days, Path outputFile, boolean obfuscate, int workers) {
        this.serverUri = serverUri;
        this.username = username;
        this.password = password;
        this.days = days;
        this.outputFile = outputFile;
        this.obfuscate = obfuscate;
        this.workers = workers;
    }

    static ExportClientSpec from(CliArgs cliArgs) {
        return new ExportClientSpec(cliArgs.server, cliArgs.username, cliArgs.password, cliArgs.days, cliArgs.outputFile, cliArgs.obfuscate, cliArgs.workers);
    }

}

File: src/main/kotlin/org/gradle/buildeng/data/collection/ExportClientSpec.kt
package org.gradle.buildeng.data.collection

import org.gradle.buildeng.app.CliArgs
import java.net.URI
import java.nio.file.Path

class ExportClientSpec constructor(val serverUri: URI, val username: String, val password: String, val days: Int, val outputFile: Path) {
    companion object {
        fun from(cliArgs: CliArgs): ExportClientSpec {
            return ExportClientSpec(cliArgs.server!!, cliArgs.username!!, cliArgs.password!!, cliArgs.days, cliArgs.outputFile!!)
        }
    }
}
